tests/pipeline/test_downsampling.py

```diff
@@ -684,3 +689,80 @@ def test_errors_on_bad_downsample_frequency(self):
             "for argument 'frequency', but got 'bad' instead."
         ).format(_qualified_name(f.downsample))
         self.assertEqual(str(e.exception), expected)
+
+
+class TestDownsampledRowwiseOperation(WithAssetFinder, ZiplineTestCase):
+    T = partial(pd.Timestamp, tz='utc')
+    START_DATE = T('2014-01-01')
+    END_DATE = T('2014-02-01')
+    HALF_WAY_POINT = T('2014-01-15')
+
+    dates = pd.date_range(START_DATE, END_DATE)
+    factor = TestingDataSet.float_col.latest
+
+    @classmethod
```

Comment:
If I put back in the old implementation, we don't actually hit the assertion error in the test; we crash in pipeline because of a shape mismatch, which I think is happening because the start and end dates specified in the test aren't trading days. I put up a change to the test that actually completes the pipeline but computes an incorrect result with the old implementation here: https://github.com/quantopian/zipline/pull/1887